airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -154,11 +154,11 @@ def __init__(
         self._slice_boundary_fields = slice_boundary_fields if slice_boundary_fields else tuple()
         self._start = start
         self._end_provider = end_provider
-        self._most_recent_record: Optional[Record] = None
         self._has_closed_at_least_one_slice = False
         self.start, self._concurrent_state = self._get_concurrent_state(stream_state)
         self._lookback_window = lookback_window
         self._slice_range = slice_range
+        self._most_recent_record_per_partition: MutableMapping[str, Record] = {}
```

Comment:
the cursor must keep track of the most recent record for each partition instead of a global value